tools/ci_build/rewriter.py

```diff
@@ -0,0 +1,226 @@
+    onnx_versioned_op = 'ONNX_OPERATOR_VERSIONED_KERNEL_CLASS_NAME'
+    onnx_versioned_op_len = len(onnx_versioned_op)
+    onnx_versioned_typed_op = 'ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME'
+    onnx_versioned_typed_op_len = len(onnx_versioned_typed_op)
+    version_map = {} #{op:{domain:[opset1, opset2, opset3 ...]}
+    operators = extract_ops_from_file(file_path, extract_ops_from_model(model_path, {}))
+
+    def fill_version_map(op_type, opset_from, opset_to, domain):
+        '''callback func to register op in version_map'''
+
+        if op_type in version_map:
+            if domain in version_map[op_type]:
+
+                if opset_from not in version_map[op_type][domain]:
+                    version_map[op_type][domain].append(opset_from)
+
+                if opset_to not in version_map[op_type][domain]:
+                    version_map[op_type][domain].append(opset_to)
+            else:
+                version_map[op_type][domain] =\
+                    [opset_from, opset_to] if opset_from < opset_to else [opset_from]
+
+        else:
+            version_map[op_type] =\
+                {domain: [opset_from, opset_to] if opset_from < opset_to else [opset_from]}
+
+        return True #end of fill_version_map(...)
+
+
+    def need_comment(op_type, opset_from, opset_to, domain):
+        '''callback func to check if the op is in ops'''
+
+        if op_type in operators and domain in operators[op_type]:
+            opset_to_index = version_map[op_type][domain].index(opset_to)
+
+            if opset_to_index == len(version_map[op_type][domain]) - 1:
+                opset_to = 9999 #if is the latest, extend to unlimited
+
+            elif opset_from == opset_to:
+                opset_to = version_map[op_type][domain][opset_to_index + 1]
+
+            for opset in range(opset_from, opset_to + 1):
+                if opset in operators[op_type][domain]:
+                    return False #do not comment
+
+        return True #end of need_comment(...)
+
+
+    def process_lines(lines, offset, end_mark, call_back):
+        '''extract op info from a logic code line start from offset to the line end
+           with end_mark, then trigger callback(op_type, opset_from, opset_to, domain)
+           return next line offset and whether current lines are disabled
+        '''
+
+        lines_to_process = []
+        while True: #collect the logical code line
+            lines_to_process.append(lines[offset])
+            stripped = lines[offset].strip()
+            if stripped.endswith(end_mark):
+                break
+            offset += 1
+        code_line = ''.join([line.strip() for line in lines_to_process])
+
+        disabled = False
+        if onnx_op in code_line:
+            trim_at = code_line.index(onnx_op) + onnx_op_len
+            args = [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            disabled = call_back(args[-1], int(args[-2]), int(args[-2]), args[-3])
+
```

Comment:
It would be helpful to put these in named variables so it's easier for someone to read the script and understand what each value is.

Reply:
Sure will do.